docs/spec/efficient_querying.rst

```diff
@@ -0,0 +1,97 @@
+.. _efficient_querying:
+
+==================
+Efficient Querying
+==================
+
+In general Kartothek should know as early as possible which data to exclude, so
+Kartothek doesn't have to read and transfer it. If you're going to query a
+dataset a lot you should consider storing it in an optimized layout first. See
+:ref:`writing`.
+
+
+Background
+----------
+
+To understand how to query efficiently you should understand Kartothek's
+querying process:
+
+When querying data, Kartothek tries to apply the given predicates as early as
+possible, so as few parquet files as possible are read and parsed:
+
+- Kartothek looks if there exists an index for the filtered column. This is
+  either if the column is partitioned (primary index) or a secondary index was
+  generated. Combine the results for multiple columns.
+- In the resulting set of Parquet files, Kartothek looks at the footer, which
+  states the range of values for each column in this file. If there are multiple
+  row groups each one specifies the range of values inside it. Decide if the
+  queried data is in the range (of a row group) and therefore if this file or
```

Comment:
Again, phrasing. `Decide` is imperative. Replace for `Kartothek checks if the queried data...`